src/main/java/org/opensearch/securityanalytics/model/STIX2IOCDto.java

```
@@ -128,8 +131,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 .field(STIX2IOC.LABELS_FIELD, labels)
                 .field(STIX2IOC.FEED_ID_FIELD, feedId)
                 .field(STIX2IOC.SPEC_VERSION_FIELD, specVersion)
-                .field(STIX2IOC.VERSION_FIELD, version)
-                .endObject();
+                .field(STIX2IOC.VERSION_FIELD, version);
+        if (numIocMatches != null) builder.field(NUM_IOC_MATCHES_FIELD, numIocMatches);
```

nit: this is fully correct but we can just add `.field(NUM_IOC_MATCHES_FIELD, numIocMatches)` even if it's null right?

---

We could, but the JSON payload would contain `"num_ioc_matches": null` as opposed to excluding that field. The Number of findings might not be necessary to provide for all uses of the DTO, so I thought it would make sense to exclude the field to avoid confusion when looking at the JSON.

I don't mind including it though.

---

good point.. can this default to 0?
also i thought this was num_ioc_matches_in_last_24h

---

Hmm would a default value of 0 be misleading in cases where the DTO is being used outside of the ListIOCs API?

---

See conversation below https://github.com/opensearch-project/security-analytics/pull/1085#discussion_r1646688097